### ISSUE TITLE
Incorrect application of comma-dangle

### ISSUE THREAD

**Issue by iptaylortechnical** [eslint/eslint#9110]
Running the ESLint plug-in on VS Code.

ESLint VS Code 1.2.11
Node 8.3
npm 4.2.0

**Please show your full configuration:**
Default

**What did you do? Please include the actual source code causing the issue.**

```js
fooBarFunction(true,
  {
    foo: 'bar',
  }
);
```
![image](https://user-images.githubusercontent.com/12981251/29297098-787c80bc-8113-11e7-8b1b-89abc37c0eca.png)

**What did you expect to happen?**
Expected no issues.

**What actually happened? Please include the actual, raw output from ESLint.**
`[eslint] Missing trailing comma. (comma-dangle)`

If you fix this issue, adding a trailing comma to the JSON parameter, it results in a 
```bash
SyntaxError: Unexpected Token )
```

 because it's the equivalent of 

```js
fooBarFunc(true, jsonParam,);
```

which is a SyntaxError. The comma-dangle rule is being misapplied to this case.

The comma-dangle rule states:

> Trailing commas simplify adding and removing items to objects and arrays, since only the lines you are modifying must be touched.

This is clearly out of scope of this rule as the JSON object in question is not a member of a larger object and therefore commas are not relevant to adding/removing. Furthermore, fixing this results in a SyntaxError.

**Comment by not-an-aardvark** [comment]
Thanks for the report.

> **Please show your full configuration:**
>
> Default

I don't think this is correct. By default, ESLint throws an error asking you to create a config file.

You can find your config by running `eslint --print-config yourFile.js`.

I suspect this is a duplicate of https://github.com/eslint/eslint/issues/7851, https://github.com/eslint/eslint/issues/8513, and https://github.com/eslint/eslint/issues/8700. You have configured the `comma-dangle` rule to require trailing function commas even though your engine doesn't support them. To fix the issue, you should either configure the parser to accept them (using `parserOptions: { ecmaVersion: 2017 }` in your config file) or reconfigure the rule to not require trailing function commas.

**Comment by iptaylortechnical** [comment]
@not-an-aardvark according to the VS Code plug-in description the plug-in `Defaults to an empty option bag.`

As for parser configuration, that is correct. From MDN: `ECMAScript 2017 allows trailing commas in function parameter lists.` Thanks for the help. Closing.
